app/services/group_assignment_service.rb

```diff
@@ -8,10 +8,17 @@ def initialize(new_group_assignment_params, new_grouping_params)
   end
 
   def build_group_assignment
-    group_assignment          = GroupAssignment.new(@new_group_assignment_params)
-    group_assignment.grouping = Grouping.where(id: @grouping_id).first_or_create!(@new_grouping_params)
+    group_assignment = GroupAssignment.new(@new_group_assignment_params)
+
+    if grouping_info_valid?
+      group_assignment.grouping = Grouping.where(id: @grouping_id).first_or_create!(@new_grouping_params)
+    end
 
     group_assignment.build_group_assignment_invitation
     group_assignment
   end
+
+  def grouping_info_valid?
+    (@grouping_id != '') || (@new_grouping_params[:title] != '')
```

Comment:
Right, I keep forgetting about `present?`. The method checks to see if either an id is present, or a title. If neither is true, we try to create a Grouping with no title which throws an exception, which is why I added this check. The method reads much better with `.present?` now